client/src/procedures/clientrunner.rs

```diff
@@ -52,22 +54,10 @@ impl Runner for Client {
     where
         F: FnOnce([Buffer<u8>; N]) -> Result<T, FatalProcedureError>,
     {
-        let mut ret = None;
-        let execute_procedure = |guard: [Buffer<u8>; N]| {
-            ret = Some(f(guard)?);
-            Ok(())
-        };
```

Comment:
What's happening here? How come this can be removed?

Reply:
Callback function `f` now returns value of type `T`, `db.get_guards` uses it to return value.